compiler/rustc_index/src/vec.rs

```diff
@@ -130,6 +137,11 @@ macro_rules! newtype_index {
                 }
             }
 
+            /// Creates a new index from a given `u32`.
+            ///
+            /// # Panics
+            ///
+            /// Will panic if `value` exceeds `MAX_AS_U32`.
```

Comment:
IIUC, `MAX` and `MAX_AS_U32` are the same value, just represented differently (one as a `u32` and one as a newtype index). So maybe both here and `from_usize` should say `MAX`?